### ISSUE TITLE
Gradient sanitizing (clipping) helps a lot :)

### ISSUE THREAD

**Issue by akanimax** [huangzh13/StyleGAN.pytorch#1]
@huangzh13,

At the outset, this is a great implementation of StyleGAN in PyTorch. I really like the way the modules are structured.
This is more of a suggestion from my side:
Seems like you are not sanitizing your gradients in the code. Please check this from the [official StyleGAN implementation](https://github.com/NVlabs/stylegan/blob/master/dnnlib/tflib/optimizer.py#L150).
I am currently working on incorporating this in my pro_gan_pytorch and BMSG-GAN.

From experience, this makes the results much better. Specifically, your generated results seem to have a few checkerboard artifacts when zoomed. These tend to disappear with gradient sanitization. 

I think the easiest way to implement this is with a backward hook in pytorch.

Hope this helps!
Again great job :clap: 

Cheers :beers:!
@akanimax

**Comment by huangzh13** [comment]
@akanimax 
Thanks for your encouragement!
I used to refer to the official tensorflow implementation, but did not notice the key points you mentioned (after all, it is hidden in tflib). Thank you!
I will continue to work hard to add more features to my implementation. 

Best,
@huangzh13
